server/src/database.js

```diff
@@ -42,7 +42,97 @@ export async function getComments(container) {
     `, [container])
 }
 
-export async function getQuestions() { return db.query('SELECT * FROM questions') }
+export async function getQuestions(container) {
+  const rows = await db.query('SELECT * FROM questions WHERE container_id = ? ORDER BY order_id, id', [container])
+  return rows.map(r => ({
+    id: r.id,
+    time: r.time,
+    text: r.text,
+    userId: r.user_id,
+    type: r.type,
+    ...JSON.parse(r.blob),
+  }))
+}
+
+function formatAnswer(r) {
+  return {
+    id: r.answer_id,
+    time: r.answer_time,
+    user: r.answer_user,
+    blob: JSON.parse(r.answer_blob),
+  }
+}
+
+function formatQuestion(rows) {
+  const r = rows[0]
+  const answers = r.answer_id ? rows.map(formatAnswer) : []
+  return {
+    id: r.question_id,
+    time: r.question_time,
+    text: r.question_text,
+    userId: r.question_user,
+    type: r.question_type,
+    order: r.question_order,
+    ...JSON.parse(r.question_blob),
+    answers,
+  }
+}
+
+export async function getQuestionsWithAnswers(container) {
```

Comment:
This could be added as a view to the database to simplify (and optimize?) so this query could be:
SELECT * FROM question_with_answer_view

Reply:
Could do this, but separate PR